components/script/dom/window.rs

```diff
@@ -2340,8 +2340,11 @@ impl Window {
             player_context,
             event_loop_waker,
         });
-
-        unsafe { WindowBinding::Wrap(JSContext::from_ptr(runtime.cx()), win) }
+        unsafe { WindowBinding::WindowWrap(
+            JSContext::from_ptr(runtime.cx()),
+            &GlobalScope::incumbent().unwrap(),
```

Comment:
Hi @jdm ! To satisfy the uniform `Wrap` function signature, I now need to pass `_scope: &GlobalScope` into `Window::Wrap` as well. It is not used in the `Wrap` function, but how to provide a "dummy" scope here? Thanks.

Reply:
This is an interesting question! I think the best thing to do here would be the following:
reflector.rs:
```rust
pub enum WrapFn {
    Global(unsafe fn(JSContext, Box<T>) -> DomRoot<T>),
    NonGlobal(unsafe fn(JSContext, &GlobalScope, Box<T>) -> DomRoot<T>)
}
```
Then the trait can use the WrapFn type and implementations can choose the appropriate variant.

Reply:
Thinking more about this, I'm not sure my solution will make a difference. I believe the problem comes from implementing DomObjectWrap as part of the dom_struct derive. If we generate it as part of the webidl codegen process instead, I believe we can choose to avoid implementing it for globals (and possibly other types that need special wrapping behaviour, like WindowProxy) and that should avoid any need for an enum like WrapFn.

Reply:
Including the trait implementation in the WebIDL codegen process seems to work! Thanks for your suggestion!